### PR TITLE
Get owned front

### PR DESCRIPTION
**Pull request by peterjah**

### CODE REVIEW

File: frontend/src/utils/write-mns-sc.tsx [hunk 1]
```diff
@@ -352,21 +356,72 @@ export function useWriteMNS(client?: Client) {
...
+            ...DOMAIN_KEY_PREFIX,
+            ...tokenIdBytes,
+          ]),
+        }),
+      );
+
+      const domainsRes = await client!
+        .publicApi()
+        .getDatastoreEntries(domainsDataStoreEntries);
+      const domains = domainsRes.map((val) => bytesToStr(val.candidate_value!));
```
Comment (modship) [inline frontend/src/utils/write-mns-sc.tsx:394]: Only 1 iteration

```ts
const targetDataStoreEntries: IDatastoreEntryInput[] = domainsRes.map((candidate) => {
  const domain = bytesToStr(candidate.candidate_value!);
  return {
    address: SC_ADDRESS,
    key: Uint8Array.from([
      ...DOMAIN_SEPARATOR_KEY,
      ...TARGET_KEY_PREFIX,
      ...strToBytes(domain),
    ]),
  };
});
```

File: frontend/src/utils/write-mns-sc.tsx [hunk 2]
```diff
@@ -352,21 +356,72 @@ export function useWriteMNS(client?: Client) {
...
+      const tokenIdsBytes = ownedKeys.map((key) => key.slice(filter.length));
+      const tokenIds = tokenIdsBytes.map((val) =>
+        bytesToU256(Uint8Array.from(val)),
+      );
+
+      const DOMAIN_SEPARATOR_KEY = [0x42];
+      const DOMAIN_KEY_PREFIX = [0x3];
+      const TARGET_KEY_PREFIX = [0x02];
+
+      const domainsDataStoreEntries: IDatastoreEntryInput[] = tokenIdsBytes.map(
```
Comment (modship) [inline frontend/src/utils/write-mns-sc.tsx:380]: Only 1 iteration if possible :
```ts
const domainsDataStoreEntries: IDatastoreEntryInput[] = tokenIdsBytes.map((tokenIdBytes) => {
  const tokenId = bytesToU256(Uint8Array.from(tokenIdBytes));
  return {
    address: SC_ADDRESS,
    key: Uint8Array.from([
      ...DOMAIN_SEPARATOR_KEY,
      ...DOMAIN_KEY_PREFIX,
      ...tokenIdBytes,
    ]),
  };
});
```

File: frontend/src/utils/write-mns-sc.tsx (hunk 1 above)
Comment (peterjah) [reply frontend/src/utils/write-mns-sc.tsx:392]: "domains" is used after. so i need to keep the full list

File: frontend/src/utils/write-mns-sc.tsx (hunk 2 above)
Comment (peterjah) [reply frontend/src/utils/write-mns-sc.tsx:378]: "tokenIds" are used after. so i need to keep the full list

**Pull request merged**